scipy/special/ellint_carlson_cpp_lite/_rj.hh

```diff
@@ -0,0 +1,701 @@
+#ifndef ELLINT_RJ_GENERIC_GUARD
+#define ELLINT_RJ_GENERIC_GUARD
+
+
+#include <algorithm>
+#include <iterator>
+#include <complex>
+#include <cmath>
+#include <cstring>
+#include "ellint_typing.hh"
+#include "ellint_argcheck.hh"
+#include "ellint_common.hh"
+#include "ellint_carlson.hh"
+
+
+/* References
+ * [1] B. C. Carlson, "Numerical computation of real or complex elliptic
+ *     integrals," Numer. Algorithm, vol. 10, no. 1, pp. 13-26, 1995.
+ *     https://arxiv.org/abs/math/9409227
+ *     https://doi.org/10.1007/BF02198293
+ * [2] B. C. Carlson, ed., Chapter 19 in "Digital Library of Mathematical
+ *     Functions," NIST, US Dept. of Commerce.
+ *     https://dlmf.nist.gov/19.20.iii
+ * [3] B. C. Carlson, J. FitzSimmons, "Reduction Theorems for Elliptic
+ *     Integrands with the Square Root of Two Quadratic Factors," J.
+ *     Comput. Appl. Math., vol. 118, nos. 1-2, pp. 71-85, 2000.
+ *     https://doi.org/10.1016/S0377-0427(00)00282-X
+ * [4] F. Johansson, "Numerical Evaluation of Elliptic Functions, Elliptic
+ *     Integrals and Modular Forms," in J. Blumlein, C. Schneider, P.
+ *     Paule, eds., "Elliptic Integrals, Elliptic Functions and Modular
+ *     Forms in Quantum Field Theory," pp. 269-293, 2019 (Cham,
+ *     Switzerland: Springer Nature Switzerland)
+ *     https://arxiv.org/abs/1806.06725
+ *     https://doi.org/10.1007/978-3-030-04480-0
+ * [5] B. C. Carlson, J. L. Gustafson, "Asymptotic Approximations for
+ *     Symmetric Elliptic Integrals," SIAM J. Math. Anls., vol. 25, no. 2,
+ *     pp. 288-303, 1994.
+ *     https://arxiv.org/abs/math/9310223
+ *     https://doi.org/10.1137/S0036141092228477
+ * [6] J. L. Gustafson, "Asymptotic Formulas for Elliptic Integrals", Doctoral
+ *     dissertataion, Dept. of Mathematics, Iowa State University, 1982
+ *     http://johngustafson.net/pubs/pub03/JGDissertation.pdf
+ */
+
+
+/* Forward declaration */
+namespace ellint_carlson
+{
+template<typename T>
+ExitStatus
+rj(const T& x, const T& y, const T& z, const T& p, const double& rerr, T& res,
+   bool noasymp = false);
+
+
+template<typename T>
+ExitStatus
+rc(const T& x, const T& y, const double& rerr, T& res);
+
+
+template<typename T>
+ExitStatus
+rg(const T& x, const T& y, const T& z, const double& rerr, T& res);
+}
+
+
+#define ELLINT_RJ_CALC()	\
```

Comment:
I would probably prefer to see these sorts of macros converted to functions, it helps when debugging and when doing static type checking, etc.  I understand if you don't want to convert all these calls though, I won't hold up this PR because of it